spring-core/src/main/java/org/springframework/core/convert/support/StringToCharacterConverter.java

```diff
@@ -28,10 +28,7 @@
 
 	@Override
 	public Character convert(String source) {
-		if (source.isEmpty()) {
```

Comment:
as far as I understand this was here because if there is an empty String, you can't represent an "empty char" (there is not such thing), thus a `null` probably makes the most sense here. I wonder if there is code that actually relies on this. 

what I mean is that someone might rely on such an un-documented specification like this:

```
StringToCharacterConverter converter = ....;
Character c = converter.convert("");
if(c == null){
     // assume "empty" and do something
}
```

the way you refactored means that every client is now broken, since you throw an `Exception`...

does this make sense? btw, I don't know what is the correct way, just adding my thoughts in here

Reply:
Thank you for your review.

I think someone would perceive null as just null, not an empty string. (I think, it is right.)
And also, the code you're talking about is a code to handle null strings, not an empty string.
Therefore, I thought it was correct to explicitly specify whether it is an empty string or null, and then intend to cope with it later. (when `Exception` throwed)

On the other hand, someone can be as intended as you say. And I sympathize with ' every client is now broken, since you throw an Exception...'.

In conclusion, this is also just my opinion. Thank you for a nice opinion. :)